src/dlite-json.h

```diff
@@ -25,8 +25,9 @@ typedef enum {
 
 /** Enum indicating whether a JSON string is formatted as data and metadata. */
 typedef enum {
-  dliteJsonDataFormat,    /*!< Data format */
-  dliteJsonMetaFormat     /*!< Metadata format */
+  dliteJsonUnknownFormat=-1,  /*!< Unknown format */
+  dliteJsonDataFormat=0,      /*!< Data format */
```

Comment:
DataFormat used to just be declared 'as-is' why do we set it to 0 now?

Reply:
This is an enum. Normally you will the compiler assigning the values instead of setting them explicit yourselves. 

However, it turns out that the compiler (at least gcc 11.3.1) by default will make DLiteJsonFormat an unsigned integer, which will mess up the negative return value of dlite_jstore_loads(). By explicitely setting `liteJsonUnknownFormat=-1` we force the compiler to make the enum a signed integer.